runtime/lua/vim/diagnostic.lua

```diff
@@ -1441,8 +1441,14 @@ M.handlers.signs = {
     local numhl = opts.signs.numhl or {}
     local linehl = opts.signs.linehl or {}
 
+    if not api.nvim_buf_is_loaded(bufnr) then
+      return
```

Comment:
is it intentional that this is done so late in the function? is that because the `text` and `opts` state needs to be updated even if the buffer isn't loaded?

Reply:
I don't think so -- I'll move it to the beginning. The `underline.show` function does this and seems to be fine